cognite/client/_api/assets.py

```diff
@@ -1012,17 +1012,14 @@ def retrieve_subtree(
     def _get_asset_subtree(self, assets: list, current_depth: int, depth: int | None) -> list:
         subtree = assets
         if depth is None or current_depth < depth:
-            children = self._get_children(assets)
-            if children:
+            if children := self._get_children(subtree):
                 subtree.extend(self._get_asset_subtree(children, current_depth + 1, depth))
         return subtree
 
     def _get_children(self, assets: list) -> list:
         ids = [a.id for a in assets]
-        tasks = []
         chunk_size = 100
-        for i in range(0, len(ids), chunk_size):
-            tasks.append({"parent_ids": ids[i : i + chunk_size], "limit": -1})
+        tasks = [{"parent_ids": chunk} for chunk in split_into_chunks(ids, chunk_size)]
```

Comment:
You lost the `"limit": -1` part it seems

Reply:
Ah yes, well spotted! Thank you! 🙏

Reply:
OK, going to push now. 😓